js/base.js

```diff
@@ -51,6 +51,13 @@ export class API {
         const topic = options.topic || name.split(".", 1)[0];
 
         await this.producer.produce(topic, event, options);
+
+        const globalDiffusion =
+            options.globalDiffusion === undefined
+                ? this.producer.globalDiffusion
+                : options.globalDiffusion;
+        const globalTopic = this._getGlobalTopic(topic);
+        if (globalDiffusion && globalTopic) await this.trigger(globalTopic, event, options);

```

Comment:
Doesn't calling itself look like a recursive loop? I guess the second time there is no globalTopic so it stops. 
Anyways, instead of `trigger` here we could use just the following right?

```js
const topic = options.topic || name.split(".", 1)[0];
const globalDiffusion =  options.globalDiffusion === undefined  ? this.producer.globalDiffusion : options.globalDiffusion;
const globalTopic = this._getGlobalTopic(topic);

await this.producer.produce(topic, event, options);
await this.producer.produce(globalTopic, event, options);
```

Reply:
This is intended. The point is to use recursion to cover cases like the one in tests `namespace1:namespace2:test-topic.test-event` in which it triggers 3 times.
Or even for real future case scenario `ripe-orchestra:failed-jobs:job-1234` for e.g. in which you could consume only logs for `ripe-orchestra-failed-jobs`.